### PR TITLE
Add ecflow support for HAFS V1

### PR DESCRIPTION
**Pull request by lgannoaa**
## Description of changes
Add the ECFLOW support for HAFSv1 operational transition. 
## Issues addressed (optional)
- fixes hafs-community/HAFS/issues/187

## Tests conducted using development environment
There are a few places contain development configuration that can not be tested with operational setup:
(1) There are a few Modules not installed in production wcoss. Therefore, used dev/para during testing.
(2) Job hafs_msg_check can not be test in development environment.
(3) The trigger event referenced to extern can not be test, because it is only available in production ecflow workflow.
(4) The time based trigger is not tested, because it needs to be adjusted under production environment.
(5) The RFC for module installation on production space has been resubmitted and the expected scheduled date for the final /apps/prod deploy will be 4/12/ for BACKUPSYS and 4/13 for PRODUCTION system. Therefore, the test will be using dev location.
Tested on WCOSS2 white space using development configuration for the following cases:
**HFSA:**
Realtime 7 cycled parallel run on JTWC 17S HERMAN storm (from 20230330 to 20230331) .
90C_2022081100
90C_2022081106
09L_2022092812
09L_2022092818
03A_2022081206
03A_2022081212
03S_2022100612
03S_2022100618
05B_2022102318
05B_2022102400
16P_2023030412
16P_2023030418
98L_2022092300 (Note, the hfsa 98L and 09L is a 5 cycled test.)
09L_2022092306
09L_2022092312
09L_2022092318
09L_2022092400
90Qfack_2022092306
90Qfack_2022092312
**HFSB**
90C_2022081100
90C_2022081106
98L_2022092300
09L_2022092306
09L_2022092312
09L_2022092318
09L_2022092400

### CODE REVIEW

File: ecf/defs/hafs.def [hunk 1]
```diff
@@ -0,0 +1,7112 @@
...
+extern /cycled_v16_3/primary/00/gfs/v16.3/gdas/atmos/post/jgdas_atmos_post_manager:release_post003
+extern /cycled_v16_3/primary/00/gfs/v16.3/gdas/atmos/post/jgdas_atmos_post_manager:release_post006
+extern /cycled_v16_3/primary/00/gfs/v16.3/gdas/atmos/post/jgdas_atmos_post_manager:release_post009
+extern /cycled_v16_3/primary/00/gfs/v16.3/gfs/atmos/post/jgfs_atmos_post_manager:release_post000
+extern /cycled_v16_3/primary/00/gfs/v16.3/gfs/atmos/post/jgfs_atmos_post_manager:release_post126
+extern /cycled_v16_3/primary/18/gfs/v16.3/gfs/atmos/post/jgfs_atmos_post_manager
+
+suite prod_hafs
+  family primary
+    edit hafs_ver 'v1.0.0'
```
Comment (BinLiu-NOAA) [inline ecf/defs/hafs.def]: @lgannoaa Could you please check if hafs_ver should be defined as v1.0 or v1.0.0? Thanks!

Comment (lgannoaa) [reply ecf/defs/hafs.def:12]: It should be set to v1.0.0. Thanks

Comment (lgannoaa) [reply ecf/defs/hafs.def]: @BinLiu-NOAA I modified ecflow workflow definition file with production extern and hafs_ver to v1.0.

**Comment by lgannoaa** [comment]
@BinLiu-NOAA, I think it is better that we let this PR to be in draft for now until the module test is completed.

**Review by BinLiu-NOAA — APPROVED** [review]
Looks good to me! Thanks a lot to @lgannoaa for developing and testing HAFSv1 Ecflow workflow and making it ready for HAFSv1 code delivery!

**Pull request merged**